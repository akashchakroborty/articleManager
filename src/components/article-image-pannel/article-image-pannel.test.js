import React from "react";
import { shallow } from "enzyme";
import ArticleImagePannel from "./article-image-pannel.component";
import {
  getReadimageContainerStyle,
  readimageContainerStyle,
  readMoreImageContainerStyle,
} from "./article-image-pannel.styles";

describe("ArticleImagePannel component", () => {
  let wrapper;
  beforeEach(() => {
    const mockProps = {
      isArticleRead: true,
    };

    wrapper = shallow(<ArticleImagePannel {...mockProps} />);
  });

  it("should render ArticleImagePannel component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should contain following styles when isArticleRead is true", () => {
    expect(
      getReadimageContainerStyle({
        isArticleRead: wrapper.props().isArticleRead,
      })
    ).toEqual(readimageContainerStyle);
  });
  it("should contain following styles when isArticleReadMore is true", () => {
    const newMockProps = {
      isArticleReadMore: true,
    };

    const newWrapper = shallow(<ArticleImagePannel {...newMockProps} />);
    expect(
      getReadimageContainerStyle({
        isArticleReadMore: newWrapper.props().isArticleReadMore,
      })
    ).toEqual(readMoreImageContainerStyle);
  });
});
