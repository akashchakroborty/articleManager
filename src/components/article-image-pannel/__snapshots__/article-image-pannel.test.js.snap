// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleImagePannel component should render ArticleImagePannel component 1`] = `ShallowWrapper {}`;
